File: Data Structures & Algorithms/boats-to-save-people/submission-1.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int start=0,end=people.size()-1;
        int count=0;
        sort(people.begin(),people.end());
        while(start<=end){
            if(people[start]+people[end]<=limit){
                count++;
                start++,end--;
                
            }
            else if(people[start]+people[end]>limit){
                count++;
                end--;
                            }
            
            
        }
        return count;
    }
};

File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-1.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        mp={}

        for i in range(len(arr)):
            mp[arr[i]]=mp.get(arr[i],0)+1

        count=0
        for i in range(len(arr)):
            if mp[arr[i]]==1:
                count+=1

                if count==k:
                    return arr[i]

        
        return ""            